src/screens/collectibles-home/collectibles-list/collectible-item/index.tsx

```diff
@@ -0,0 +1,89 @@
+import BigNumber from 'bignumber.js';
+import React, { memo, useMemo } from 'react';
+import { StyleProp, Text, View, ViewStyle } from 'react-native';
+import { TouchableOpacity } from 'react-native-gesture-handler';
+
+import { objktCurrencies } from 'src/apis/objkt';
+import { ModalsEnum } from 'src/navigator/enums/modals.enum';
+import { useNavigation } from 'src/navigator/hooks/use-navigation.hook';
+import {
+  useCollectibleDetailsLoadingSelector,
+  useCollectibleDetailsSelector
+} from 'src/store/collectibles/collectibles-selectors';
+import { TokenInterface } from 'src/token/interfaces/token.interface';
+import { formatNumber } from 'src/utils/format-price';
+import { isDefined } from 'src/utils/is-defined';
+import { mutezToTz } from 'src/utils/tezos.util';
+
+import { Balance } from './balance';
+import { CollectibleItemImage } from './item-image';
+import { useCollectibleItemStyles } from './styles';
+
+interface Props {
+  slug: string;
+  collectible: TokenInterface;
+  size: number;
+  isShowInfo?: boolean;
+  style?: StyleProp<ViewStyle>;
+}
+
+export const CollectibleItem = memo<Props>(({ slug, collectible, size, isShowInfo = false, style }) => {
+  const { navigate } = useNavigation();
+
+  const handleNavigate = () => navigate(ModalsEnum.CollectibleModal, { slug });
+
+  const styles = useCollectibleItemStyles();
+
+  const areDetailsLoading = useCollectibleDetailsLoadingSelector();
+  const details = useCollectibleDetailsSelector(slug);
+
+  const balance = collectible.balance;
+
+  const listing = useMemo(() => {
+    const cheapestListing = details?.listingsActive[0];
+    if (!cheapestListing) {
+      return null;
+    }
+
+    const { price, currency_id } = cheapestListing;
+
+    const currency = objktCurrencies[currency_id];
+    if (!isDefined(currency)) {
+      return null;
+    }
+
+    const floorPrice = +mutezToTz(new BigNumber(price), currency.decimals);
```

Comment:
Maybe let's not use `+` for converting `BigNumber` to `number`. It's better to use `.toNumber()`